a/asttokens/asttokens_ubi_8.4.sh

```diff
@@ -0,0 +1,71 @@
+# ----------------------------------------------------------------------------
+#
+# Package       : asttokens
+# Version       : v2.0.5
+# Source repo   : https://github.com/gristlabs/asttokens
+# Tested on     : UBI 8.4
+# Script License: Apache License, Version 2 or later
+# Maintainer    : Gururaj R Katti <Gururaj.Katti@ibm.com>
+#
+# Disclaimer: This script has been tested in non-root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+#!/bin/bash
+
+export ASTTOKENS_SLOW_TESTS=1
+
+
+if [ -z "$1" ]; then
+  export VERSION=master
```

Comment:
Should we have `v2.0.5` here as default? Since there is an option to pass another version/branch as parameter, which can server for `master`, etc.

Reply:
1. With invalid argument

testw@38ae6b908008 ~]$ ./asttokens_ubi_8.4.sh v2.6.0
Updating Subscription Management repositories.
Unable to read consumer identity

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.

Last metadata expiration check: 1 day, 0:24:25 ago on Mon Oct 11 07:41:40 2021.
Package python39-3.9.2-1.module+el8.4.0+10237+bdc77aac.ppc64le is already installed.
Package git-2.27.0-1.el8.ppc64le is already installed.
Package wget-1.19.5-10.el8.ppc64le is already installed.
Package gcc-8.4.1-1.el8.ppc64le is already installed.
Package python39-devel-3.9.2-1.module+el8.4.0+10237+bdc77aac.ppc64le is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Cloning into 'asttokens'...
remote: Enumerating objects: 708, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 708 (delta 13), reused 17 (delta 6), pack-reused 676
Receiving objects: 100% (708/708), 179.90 KiB | 1.41 MiB/s, done.
Resolving deltas: 100% (466/466), done.
error: pathspec 'v2.6.0' did not match any file(s) known to git

2. With valid argument

[testw@38ae6b908008 ~]$ ./asttokens_ubi_8.4.sh master
Updating Subscription Management repositories.
Unable to read consumer identity

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.

Last metadata expiration check: 1 day, 0:25:24 ago on Mon Oct 11 07:41:40 2021.
Package python39-3.9.2-1.module+el8.4.0+10237+bdc77aac.ppc64le is already installed.
Package git-2.27.0-1.el8.ppc64le is already installed.
Package wget-1.19.5-10.el8.ppc64le is already installed.
Package gcc-8.4.1-1.el8.ppc64le is already installed.
Package python39-devel-3.9.2-1.module+el8.4.0+10237+bdc77aac.ppc64le is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Cloning into 'asttokens'...
remote: Enumerating objects: 708, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 708 (delta 13), reused 17 (delta 6), pack-reused 676
Receiving objects: 100% (708/708), 179.90 KiB | 1006.00 KiB/s, done.
Resolving deltas: 100% (466/466), done.
Already on 'master'
Your branch is up to date with 'origin/master'.
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: coveralls in /home/testw/.local/lib/python3.9/site-packages (3.2.0)
Requirement already satisfied: docopt>=0.6.1 in /home/testw/.local/lib/python3.9/site-packages (from coveralls) (0.6.2)
Requirement already satisfied: requests>=1.0.0 in /home/testw/.local/lib/python3.9/site-packages (from coveralls) (2.26.0)
Requirement already satisfied: coverage<6.0,>=4.1 in /home/testw/.local/lib/python3.9/site-packages (from coveralls) (5.5)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (1.26.6)
Requirement already satisfied: idna<4,>=2.5 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (3.2)
Requirement already satisfied: certifi>=2017.4.17 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (2021.5.30)
Requirement already satisfied: charset-normalizer~=2.0.0 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (2.0.4)
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/asttokens
copying asttokens/util.py -> build/lib/asttokens
copying asttokens/mark_tokens.py -> build/lib/asttokens
copying asttokens/version.py -> build/lib/asttokens
copying asttokens/line_numbers.py -> build/lib/asttokens
copying asttokens/__init__.py -> build/lib/asttokens
copying asttokens/asttokens.py -> build/lib/asttokens
installing to build/bdist.linux-ppc64le/wheel
running install
running install_lib
creating build/bdist.linux-ppc64le
creating build/bdist.linux-ppc64le/wheel
creating build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/util.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/mark_tokens.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/version.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/line_numbers.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/__init__.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/asttokens.py -> build/bdist.linux-ppc64le/wheel/asttokens
running install_egg_info
running egg_info
creating asttokens.egg-info
writing asttokens.egg-info/PKG-INFO
writing dependency_links to asttokens.egg-info/dependency_links.txt
writing requirements to asttokens.egg-info/requires.txt
writing top-level names to asttokens.egg-info/top_level.txt
writing manifest file 'asttokens.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE'
writing manifest file 'asttokens.egg-info/SOURCES.txt'
Copying asttokens.egg-info to build/bdist.linux-ppc64le/wheel/asttokens-2.0.6.dev1+g1bf641c-py3.9.egg-info
running install_scripts
creating build/bdist.linux-ppc64le/wheel/asttokens-2.0.6.dev1+g1bf641c.dist-info/WHEEL
creating 'dist/asttokens-2.0.6.dev1+g1bf641c-py2.py3-none-any.whl' and adding 'build/bdist.linux-ppc64le/wheel' to it
adding 'asttokens/__init__.py'
adding 'asttokens/asttokens.py'
adding 'asttokens/line_numbers.py'
adding 'asttokens/mark_tokens.py'
adding 'asttokens/util.py'
adding 'asttokens/version.py'
adding 'asttokens-2.0.6.dev1+g1bf641c.dist-info/LICENSE'
adding 'asttokens-2.0.6.dev1+g1bf641c.dist-info/METADATA'
adding 'asttokens-2.0.6.dev1+g1bf641c.dist-info/WHEEL'
adding 'asttokens-2.0.6.dev1+g1bf641c.dist-info/top_level.txt'
adding 'asttokens-2.0.6.dev1+g1bf641c.dist-info/RECORD'
removing build/bdist.linux-ppc64le/wheel
Processing ./dist/asttokens-0.CI-py2.py3-none-any.whl
Requirement already satisfied: six in /usr/local/lib/python3.9/site-packages (from asttokens==0.CI) (1.16.0)
Requirement already satisfied: astroid in /usr/local/lib/python3.9/site-packages (from asttokens==0.CI) (2.7.3)
Requirement already satisfied: pytest in /usr/local/lib/python3.9/site-packages (from asttokens==0.CI) (6.2.5)
Requirement already satisfied: setuptools>=20.0 in /usr/lib/python3.9/site-packages (from astroid->asttokens==0.CI) (50.3.2)
Requirement already satisfied: lazy-object-proxy>=1.4.0 in /usr/local/lib64/python3.9/site-packages (from astroid->asttokens==0.CI) (1.6.0)
Requirement already satisfied: wrapt<1.13,>=1.11 in /usr/local/lib64/python3.9/site-packages (from astroid->asttokens==0.CI) (1.12.1)
Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (1.0.0)
Requirement already satisfied: py>=1.8.2 in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (1.10.0)
Requirement already satisfied: packaging in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (21.0)
Requirement already satisfied: iniconfig in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (1.1.1)
Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (21.2.0)
Requirement already satisfied: toml in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (0.10.2)
Requirement already satisfied: pyparsing>=2.0.2 in /usr/local/lib/python3.9/site-packages (from packaging->pytest->asttokens==0.CI) (3.0.0rc1)
Installing collected packages: asttokens
  Attempting uninstall: asttokens
    Found existing installation: asttokens 2.0.5
    Uninstalling asttokens-2.0.5:
      Successfully uninstalled asttokens-2.0.5
Successfully installed asttokens-2.0.6.dev1+g1bf641c
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /home/testw/asttokens, configfile: setup.cfg
plugins: cov-2.12.1
collected 107 items

tests/test_astroid.py .............s.................................    [ 43%]
tests/test_asttokens.py ......                                           [ 49%]
tests/test_line_numbers.py ...                                           [ 52%]
tests/test_mark_tokens.py .............................................. [ 95%]
.                                                                        [ 96%]
tests/test_util.py ....                                                  [100%]

------------- generated xml file: /home/testw/asttokens/rspec.xml --------------
=========== 106 passed, 1 skipped, 2 warnings in 1366.95s (0:22:46) ============
[testw@38ae6b908008 ~]$

3. Without argument

[testw@38ae6b908008 ~]$ ./asttokens_ubi_8.4.sh
Updating Subscription Management repositories.
Unable to read consumer identity

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.

Last metadata expiration check: 1 day, 2:03:55 ago on Mon Oct 11 07:41:40 2021.
Package python39-3.9.2-1.module+el8.4.0+10237+bdc77aac.ppc64le is already installed.
Package git-2.27.0-1.el8.ppc64le is already installed.
Package wget-1.19.5-10.el8.ppc64le is already installed.
Package gcc-8.4.1-1.el8.ppc64le is already installed.
Package python39-devel-3.9.2-1.module+el8.4.0+10237+bdc77aac.ppc64le is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Cloning into 'asttokens'...
remote: Enumerating objects: 708, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 708 (delta 13), reused 17 (delta 6), pack-reused 676
Receiving objects: 100% (708/708), 179.90 KiB | 1.31 MiB/s, done.
Resolving deltas: 100% (466/466), done.
Note: switching to 'v2.0.5'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at fb47539 Bump patch version
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: coveralls in /home/testw/.local/lib/python3.9/site-packages (3.2.0)
Requirement already satisfied: coverage<6.0,>=4.1 in /home/testw/.local/lib/python3.9/site-packages (from coveralls) (5.5)
Requirement already satisfied: requests>=1.0.0 in /home/testw/.local/lib/python3.9/site-packages (from coveralls) (2.26.0)
Requirement already satisfied: docopt>=0.6.1 in /home/testw/.local/lib/python3.9/site-packages (from coveralls) (0.6.2)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (1.26.6)
Requirement already satisfied: certifi>=2017.4.17 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (2021.5.30)
Requirement already satisfied: charset-normalizer~=2.0.0 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /home/testw/.local/lib/python3.9/site-packages (from requests>=1.0.0->coveralls) (3.2)
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/asttokens
copying asttokens/util.py -> build/lib/asttokens
copying asttokens/mark_tokens.py -> build/lib/asttokens
copying asttokens/version.py -> build/lib/asttokens
copying asttokens/line_numbers.py -> build/lib/asttokens
copying asttokens/__init__.py -> build/lib/asttokens
copying asttokens/asttokens.py -> build/lib/asttokens
installing to build/bdist.linux-ppc64le/wheel
running install
running install_lib
creating build/bdist.linux-ppc64le
creating build/bdist.linux-ppc64le/wheel
creating build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/util.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/mark_tokens.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/version.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/line_numbers.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/__init__.py -> build/bdist.linux-ppc64le/wheel/asttokens
copying build/lib/asttokens/asttokens.py -> build/bdist.linux-ppc64le/wheel/asttokens
running install_egg_info
running egg_info
creating asttokens.egg-info
writing asttokens.egg-info/PKG-INFO
writing dependency_links to asttokens.egg-info/dependency_links.txt
writing requirements to asttokens.egg-info/requires.txt
writing top-level names to asttokens.egg-info/top_level.txt
writing manifest file 'asttokens.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE'
writing manifest file 'asttokens.egg-info/SOURCES.txt'
Copying asttokens.egg-info to build/bdist.linux-ppc64le/wheel/asttokens-2.0.5-py3.9.egg-info
running install_scripts
creating build/bdist.linux-ppc64le/wheel/asttokens-2.0.5.dist-info/WHEEL
creating 'dist/asttokens-2.0.5-py2.py3-none-any.whl' and adding 'build/bdist.linux-ppc64le/wheel' to it
adding 'asttokens/__init__.py'
adding 'asttokens/asttokens.py'
adding 'asttokens/line_numbers.py'
adding 'asttokens/mark_tokens.py'
adding 'asttokens/util.py'
adding 'asttokens/version.py'
adding 'asttokens-2.0.5.dist-info/LICENSE'
adding 'asttokens-2.0.5.dist-info/METADATA'
adding 'asttokens-2.0.5.dist-info/WHEEL'
adding 'asttokens-2.0.5.dist-info/top_level.txt'
adding 'asttokens-2.0.5.dist-info/RECORD'
removing build/bdist.linux-ppc64le/wheel
Processing ./dist/asttokens-0.CI-py2.py3-none-any.whl
Requirement already satisfied: six in /usr/local/lib/python3.9/site-packages (from asttokens==0.CI) (1.16.0)
Requirement already satisfied: pytest in /usr/local/lib/python3.9/site-packages (from asttokens==0.CI) (6.2.5)
Requirement already satisfied: astroid in /usr/local/lib/python3.9/site-packages (from asttokens==0.CI) (2.7.3)
Requirement already satisfied: wrapt<1.13,>=1.11 in /usr/local/lib64/python3.9/site-packages (from astroid->asttokens==0.CI) (1.12.1)
Requirement already satisfied: setuptools>=20.0 in /usr/lib/python3.9/site-packages (from astroid->asttokens==0.CI) (50.3.2)
Requirement already satisfied: lazy-object-proxy>=1.4.0 in /usr/local/lib64/python3.9/site-packages (from astroid->asttokens==0.CI) (1.6.0)
Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (21.2.0)
Requirement already satisfied: py>=1.8.2 in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (1.10.0)
Requirement already satisfied: packaging in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (21.0)
Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (1.0.0)
Requirement already satisfied: iniconfig in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (1.1.1)
Requirement already satisfied: toml in /usr/local/lib/python3.9/site-packages (from pytest->asttokens==0.CI) (0.10.2)
Requirement already satisfied: pyparsing>=2.0.2 in /usr/local/lib/python3.9/site-packages (from packaging->pytest->asttokens==0.CI) (3.0.0rc1)
Installing collected packages: asttokens
  Attempting uninstall: asttokens
    Found existing installation: asttokens 2.0.6.dev1+g1bf641c
    Uninstalling asttokens-2.0.6.dev1+g1bf641c:
      Successfully uninstalled asttokens-2.0.6.dev1+g1bf641c
Successfully installed asttokens-2.0.5
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /home/testw/asttokens, configfile: setup.cfg
plugins: cov-2.12.1
collected 107 items

tests/test_astroid.py .............s.................................    [ 43%]
tests/test_asttokens.py ......                                           [ 49%]
tests/test_line_numbers.py ...                                           [ 52%]
tests/test_mark_tokens.py .............................................. [ 95%]
.                                                                        [ 96%]
tests/test_util.py ....                                                  [100%]

------------- generated xml file: /home/testw/asttokens/rspec.xml --------------
=========== 106 passed, 1 skipped, 2 warnings in 1377.21s (0:22:57) ============
[testw@38ae6b908008 ~]$